shopfloor/actions/search.py

```diff
@@ -168,12 +168,18 @@ def generic_packaging_from_scan(self, barcode):
             [("barcode", "=", barcode), ("product_id", "=", False)], limit=1
         )
 
+    def _get_origin_from_barcode(self, barcode):
+        """Returns a list of valid origins for a given barcode."""
+        # Hookable method
+        return [barcode]
+
     def origin_move_from_scan(self, barcode, extra_domain=None):
+        origins = self.env["delivery.carrier"]._get_origin_from_barcode(barcode)
```

Comment:
You probably wanted to write this:
```suggestion
        origins = self._get_origin_from_barcode(barcode)
```